Documentation/Authentication.md

```diff
@@ -1,140 +1,119 @@
+## Service Principal Authentication
+
+Interactive login, similar to how the CLI authenticates, is the easiest way to
+authenticate; however, when using the Node.js SDK programmatically, you will need
+to use service principal authentication. This essentially creates keys for your
+Azure Active Directory account that you can provide to the SDK to authenticate
+rather than requiring user intervention or username/password.
+
+### Creating a Service Principal
+
+There are two ways to create a Service Principal, the next sections will walk you
+through each method.
+
+#### 1. Azure Portal
 
-#### Step 1. Create a service principal
+Follow the steps outlined in the
+[Azure Portal documentation](https://azure.microsoft.com/en-us/documentation/articles/resource-group-create-service-principal-portal/)
+ to generate the necessary keys.
 
-Let us create a serviceprincipal with a password. The default start-date will be the day of application creation and the default end-date will be 1 year from the day of creation.
+#### 2. Azure CLI
 
-> ###### NOTE: The AppId a.k.a the serviceprincipal name in **GUID** format that gets created in this command is also known as clientId. It is the id given by ADAL for itself to identify your application. You can use this as the clientId during user authentication (creating UserTokenCredentials object).
+This method can be used with either the
+[Azure Cross-Platform CLI (npm module)](https://github.com/Azure/azure-xplat-cli)
+or the
+[Azure CLI v2.0 (Python)](https://github.com/Azure/azure-cli).
 
+_Using the Node.js cross-platform CLI_
+```shell
+$ azure login # or $ azure login -u user@domain.tld
+$ azure ad sp create -n sp-name -p sp-password
 ```
-D:\sdk>azure ad sp create -n testap908 -p P@ssw0rd
-info:    Executing command ad sp create
-+ Creating application testap674
-+ Creating service principal for application 56894bd4-0fde-41d8-a0d7-5bsslccety2
+
+This will create a new Service Principal and output the keys, copy the output for
+use in your script. Note: you can retrieve the keys later by running
+`$ azure ad sp list` in your terminal.
+
+The important fields are marked below (the other required field is the password
+that was provided when creating the service principal)
+
+```shell
++ Creating application sp-name
++ Creating service principal for application **56894bd4-0fde-41d8-a0d7-5bsslccety2**
 data:    Object Id:               weewrerer-e329-4e9b-98c6-7878787
-data:    Display Name:            testap674
+data:    Display Name:            sp-name
 data:    Service Principal Names:
-data:                             56894bd4-0fde-41d8-a0d7-5bsslccety2   <<<<<<<<<<<<<<<<<<<<< also known as clientId
-data:                             https://testap908
+data:                             **56894bd4-0fde-41d8-a0d7-5bsslccety2**
+data:                             https://sp-name
 info:    ad sp create command OK
 ```
 
-#### Step 2. Assigning a role to a ServicePrincipal by using the spn
```

Comment:
Don't you need role assignment? If you still need to do this, I'd prefer to have it in the walkthrough.